volume/volume.go

```diff
@@ -104,8 +110,9 @@ type MountPoint struct {
 	CopyData bool `json:"-"`
 	// ID is the opaque ID used to pass to the volume driver.
 	// This should be set by calls to `Mount` and unset by calls to `Unmount`
-	ID   string `json:",omitempty"`
-	Spec mounttypes.Mount
+	ID    string `json:",omitempty"`
+	Tmpfs *Tmpfs `json:"-"`
```

Comment:
For passing the spec to the `mount(2)` syscall, we need to compile the typed spec struct to string in somewhere.

I can come up with an alternative that defers such compilation, but I think it is complicated

alternative `volume/volume.go`:
```go
import (
        mounttypes "github.com/docker/docker/api/types/mount"
)

type Tmpfs struct {
  Options *mounttypes.TmpfsOptions // typed spec, for `--mount`
  RawOptions *string // raw string, for old `--tmpfs`
}
```

alternative `container/container_unix.go`:
```go
func (container *Container) TmpfsMounts() ([]Mount, error) {
	var mounts []Mount
	for dest, mnt := range container.MountPoints {
		if mnt.Type == mounttypes.TypeTmpfs {
			data := ""
			if mnt.Tmpfs != nil {
				if mnt.Tmpfs.RawOptions == nil && if mnt.Tmpfs.Options == nil {
					return nil, fmt.Errorf("some error")
				}
				if mnt.Tmpfs.RawOptions != nil && if mnt.Tmpfs.Options != nil {
					return nil, fmt.Errorf("some error")
				}			
			        if mnt.Tmpfs.RawOptions != nil {
					data = mnt.Tmpfs.RawOptions
				} else if mnt.Tmpfs.Options != nil {
					data = ConvertTmpfsOptions(mnt.Tmpfs.Options)
				}
			}
			mounts = append(mounts, Mount{
				Source:      "tmpfs",
				Destination: dest,
				Data:        data,
			})
		}
	}

	return mounts, nil
}
```